interfaces/repo.go

```diff
@@ -80,6 +80,112 @@ func (r *Repository) AddInterface(i Interface) error {
 	return nil
 }
 
+// QueryOptions describes options for QueryInterfaces.
+//
+// Names: return just this subset if non-empty.
+// Doc: return documentation.
+// Plugs: return information about plugs.
+// Slots: return information about slots.
+// Connected: only consider interfaces with at least one connection.
+type QueryOptions struct {
+	Names                        []string
+	Doc, Plugs, Slots, Connected bool
+}
+
+func (r *Repository) queryInterface(iface Interface, opts *QueryOptions) *InterfaceInfo {
+	// NOTE: QueryOptions.Connected is handled by QueryInterfaces
+	md := MetaDataOf(iface)
```

Comment:
I like StaticInfo, it was conveying the "bag of stuff baked into the binary" idea I had before better than the generic meta-data.